plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -1231,12 +1236,13 @@ private void executeExpireSnapshots(ConnectorSession session, IcebergTableExecut
         long expireTimestampMillis = session.getStart().toEpochMilli() - retention.toMillis();
         TrinoFileSystem fileSystem = fileSystemFactory.create(session);
         List<String> pathsToDelete = new ArrayList<>();
+        AtomicLong failedToDelete = new AtomicLong();
```

Comment:
This doesn't have to be thread safe, if that delete function was called from multi thread it probably wouldn't work as `pathsToDelete` is not thread safe.